BootloaderCorePkg/Library/SmbiosInitLib/SmbiosInitLib.c

```diff
@@ -379,6 +405,42 @@ AddSmbiosType (
 }
 
 
+/**
+  This function is called to initialize the SmbiosStringsPtr.
+**/
+VOID
+EFIAPI
+InitSmbiosStringPtr (
+  VOID
+  )
+{
+
+  EFI_STATUS            Status;
+  UINT16                Length;
+  VOID                 *SmbiosStringsPtr;
+
+  //
+  // Allocate the memmory for SMBIOS String Ptr,
+  // Copy contents from Default String Array
+  //
+
+  Length = sizeof (DefaultSmbiosStrings) / sizeof (SMBIOS_TYPE_STRINGS);

```

Comment:
Can we use this PcdSmbiosStringsCnt to indicate how many entry to allocate ?   The ideal is to allow platform to modify and append new entries.  If it is the same size as DefaultSmbiosStrings, then it has no room to append.  The initial value for PcdSmbiosStringsCnt  needs to be greater then the entry in DefaultSmbiosStrings.